spec/lib/efile/nj/nj_retirement_income_helper_spec.rb

```diff
@@ -0,0 +1,132 @@
+      { traits: [:married_filing_separately], total_income: 120_000, expected: 30_000 },
+
+      { total_income: 130_000, expected: 24_375 },
+      { traits: [:head_of_household], total_income: 130_000, expected: 24_375 },
+      { traits: [:qualifying_widow], total_income: 130_000, expected: 24_375 },
+      { traits: [:married_filing_jointly], total_income: 130_000, expected: 32_500 },
+      { traits: [:married_filing_separately], total_income: 130_000, expected: 16_250 },
+
+      { total_income: 150_001, expected: 0 },
+      { traits: [:head_of_household], total_income: 150_001, expected: 0 },
+      { traits: [:qualifying_widow], total_income: 150_001, expected: 0 },
+      { traits: [:married_filing_jointly], total_income: 150_001, expected: 0 },
+      { traits: [:married_filing_separately], total_income: 150_001, expected: 0 },
+    ].each do |test_case|
+      context "when filing with #{test_case}" do
+        let(:intake) do
+          create(:state_file_nj_intake, *test_case[:traits])
+        end
+        it "returns #{test_case[:expected]}" do
+          helper = Efile::Nj::NjRetirementIncomeHelper.new(intake)
+          result = helper.calculate_maximum_exclusion(test_case[:total_income])
+          expect(result).to eq(test_case[:expected])
+        end
+      end
+    end
+
+    describe ".eligible?" do
+      [
+        { traits: [:mfj_spouse_over_65], total_income: 99_000, earned_income: 3000, line_28a: 0, expected: true },
+        { traits: [:primary_over_65], total_income: 99_000, earned_income: 3000, line_28a: 0, expected: true },
+        { total_income: 99_000, earned_income: 3000, line_28a: 0, expected: false },
+        { traits: [:primary_over_65], total_income: 160_000, earned_income: 3000, line_28a: 0, expected: false },
+        { traits: [:primary_over_65], total_income: 99_000, earned_income: 4000, line_28a: 0, expected: false },
+        { traits: [:primary_over_65], total_income: 99_000, earned_income: 4000, line_28a: 100_000, expected: false },
+      ].each do |test_case|
+        context "when filing with #{test_case}" do
+          let(:intake) do
+            create(:state_file_nj_intake, *test_case[:traits])
+          end
+          let(:instance) do
+            Efile::Nj::Nj1040Calculator.new(
+              year: MultiTenantService.statefile.current_tax_year,
+              intake: intake
+            )
+          end
+          before do
+            allow_any_instance_of(Efile::Nj::Nj1040Calculator).to receive(:calculate_line_15).and_return(test_case[:earned_income])
+            allow_any_instance_of(Efile::Nj::Nj1040Calculator).to receive(:calculate_line_27).and_return(test_case[:total_income])
+            allow_any_instance_of(Efile::Nj::Nj1040Calculator).to receive(:calculate_line_28a).and_return(test_case[:line_28a])
+            instance.calculate
+          end
+  
+          it "returns #{test_case[:expected]}" do
+            helper = Efile::Nj::NjRetirementIncomeHelper.new(intake)
```

Comment:
[dust] can we instantiate the helper in a before each (and still make use of lazy definition for the intake as-is) so that we don't have to duplicate this in every test?